doc/modules/mod_global_distrib.md

```diff
@@ -0,0 +1,233 @@
+### Module Description
+
+> **Experimental:** we welcome suggestions and bug reports but do not recommend enabling `mod_global_distrib` on a production service.
+
+This module enables global distribution of a single XMPP domain.
+With `mod_global_distrib`, multiple distinct MongooseIM clusters can share a single domain name and route messages to the specific datacenter where the recipient is available.
+
+### How it works
+
+There are multiple subsystems that cooperate to enable global distribution:
+
+#### Metadata sharing
+
+Sharing of metadata is done by leveraging a database with cross-datacenter replication.
+Currently, only Redis is supported, with [Dynomite](https://github.com/Netflix/dynomite) layer for replication.
+The most important metadata stored in the database is a **session/routing table**.
+The table stores mappings between currently logged users' JIDs and datacenters on which they are logged in.
+Because access to the session table is very frequent, its entries are additionally cached on each node.
+
+To preserve consistency between database instances, all data is stored with a set expiration time and is periodically refreshed.
+Each node of each cluster is responsible for refreshing its own data.
+Thus, in an event of a netsplit datacenters will have their information about other datacenter's users expire, as those users are now unreachable; but once the connection is reestablished, the data will be replicated again as datacenters refresh their entries.
```

Comment:
Multiple datacenters, right?
If so:

Thus, in an event of a netsplit, datacenters will have their information about other datacenters' users expire, as those users are now unreachable; but once the connection is reestablished, the data will be replicated again as datacenters refresh their entries.